lib/go/edgecontext/backend.go

```diff
@@ -0,0 +1,124 @@
+package edgecontext
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	v0 "github.com/reddit/edgecontext/lib/go/internal/v0"
+)
+
+type Backend struct {
+	SetEdgeContext func(context.Context, *EdgeRequestContext) context.Context
+	GetEdgeContext func(context.Context) (*EdgeRequestContext, bool)
+	Factory        func(Config) ecinterface.Factory
+	Init           func(Config) *Impl
+	New            func(context.Context, *Impl, NewArgs) (*EdgeRequestContext, error)
+	FromHeader     func(context.Context, string, *Impl) (*EdgeRequestContext, error)
+}
+
+var defaultBackend = &Backend{
+	SetEdgeContext: func(ctx context.Context, ec *EdgeRequestContext) context.Context {
+		if ec == nil {
+			return ctx
+		}
+		return v0.SetEdgeContext(ctx, ec.Source.(*v0.DataSource))
+	},
+	GetEdgeContext: func(ctx context.Context) (*EdgeRequestContext, bool) {
+		source, ok := v0.GetEdgeContext(ctx)
+		if !ok {
+			return nil, false
+		}
+		return &EdgeRequestContext{
+			Source: source,
+			ctx:    ctx,
+		}, true
+	},
+	Factory: func(cfg Config) ecinterface.Factory {
+		return v0.Factory(v0.Config{
+			Store:  cfg.Store,
+			Logger: cfg.Logger,
+		})
+	},
+	Init: func(cfg Config) *Impl {
+		return &Impl{
+			Interface: v0.Init(v0.Config{
+				Store:  cfg.Store,
+				Logger: cfg.Logger,
+			}),
+		}
+	},
+	New: func(ctx context.Context, impl *Impl, args NewArgs) (*EdgeRequestContext, error) {
+		unwrapped, ok := impl.Interface.(*v0.Impl)
+		if !ok {
+			return nil, fmt.Errorf("unwrapped interface is not a v0.Impl")
+		}
+		source, err := v0.New(
+			ctx,
+			unwrapped,
+			v0.NewArgs{
+				LoID:              args.LoID,
+				LoIDCreatedAt:     args.LoIDCreatedAt,
+				SessionID:         args.SessionID,
+				DeviceID:          args.DeviceID,
+				AuthToken:         args.AuthToken,
+				OriginServiceName: args.OriginServiceName,
+				CountryCode:       args.CountryCode,
+				RequestID:         args.RequestID,
+				LocaleCode:        args.LocaleCode,
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("creating v0 data Source: %w", err)
+		}
+		return NewFromSource(ctx, source), nil
+	},
+	FromHeader: func(ctx context.Context, header string, impl *Impl) (*EdgeRequestContext, error) {
+		unwrapped, ok := impl.Interface.(*v0.Impl)
+		if !ok {
+			return nil, fmt.Errorf("unwrapped interface is not a v0.Impl")
+		}
+		source, err := v0.FromHeader(ctx, header, unwrapped)
+		if err != nil {
+			return nil, fmt.Errorf("creating v0 data Source: %w", err)
+		}
+		if source == nil {
+			return nil, nil
+		}
+		return NewFromSource(ctx, source), nil
+	},
+}
+
+var backend *Backend
```

Comment:
use `atomic.Pointer[Backend]` instead? then you can add an `init`:
```go
func init() {
  backend.Store(defaultBackend)
}
```
and this way it would never be nil (you can also just do `var backend *Backend = defaultBackend` so it also would never be nil, if you don't think concurrency-safe is needed)